packages/components/src/components/Dialog/index.tsx

```diff
@@ -0,0 +1,179 @@
+
+import css from '@styled-system/css';
+import { createGlobalStyle } from 'styled-components';
+import { AnimatePresence, motion } from 'framer-motion';
+
+import Rect, { useRect } from '@reach/rect';
+import { DialogOverlay, DialogContent } from '@reach/dialog';
+import '@reach/dialog/styles.css';
+
+import { Button } from '../Button';
+import { IconButton } from '../IconButton';
+
+const GlobalStyle = createGlobalStyle(
+  css({
+    '[data-reach-dialog-overlay][data-component="DialogOverlay"]': {
+      background: 'transparent',
+    },
+    '[data-reach-dialog-content][data-component="DialogContent"]': {
+      backgroundColor: 'dialog.background',
+      color: 'dialog.foreground',
+      border: '1px solid',
+      borderColor: 'dialog.border',
+      borderRadius: 3,
+      boxShadow: 2,
+      position: 'absolute',
+      // override reach styles
+      padding: 0,
+      margin: 0,
+      width: 'auto',
+    },
+  })
+);
+
+const context = React.createContext({
+  dialogVisible: false,
+  setDialogVisibility: null,
+  triggerRef: null,
+  triggerRect: null,
+  label: null,
+});
+
+type DialogTypes = React.FC<{
+  /** Accessible label for dialog content */
+  label: string;
+}> & {
+  Button: typeof DialogButton;
+  IconButton: typeof DialogIconButton;
+  Content: typeof Content;
+};
+
+const Dialog: DialogTypes = ({ label, children }) => {
+  const [dialogVisible, setDialogVisibility] = React.useState(false);
+  const triggerRef = React.useRef();
+  const triggerRect = useRect(triggerRef);
+
+  return (
+    <context.Provider
+      value={{
+        dialogVisible,
+        setDialogVisibility,
+        triggerRef,
+        triggerRect,
+        label,
+      }}
+    >
+      <GlobalStyle />
+      {children}
+    </context.Provider>
+  );
+};
+
+const DialogButton = props => {
+  const { setDialogVisibility, triggerRef } = React.useContext(context);
+
+  return (
+    <Button
+      {...props}
+      css={deepmerge({ width: 'auto' }, props.css || {})}
+      ref={triggerRef}
+      onClick={() => setDialogVisibility(true)}
+    />
+  );
+};
+
+const DialogIconButton = props => {
+  const { setDialogVisibility, triggerRef, label } = React.useContext(context);
+
+  return (
+    <IconButton
+      {...props}
+      title={props.label || label}
+      css={deepmerge({ width: '26px' }, props.css || {})}
+      innerRef={triggerRef}
+      onClick={() => setDialogVisibility(true)}
+    />
+  );
+};
+
+const Content = ({ style = {}, children, ...props }) => {
+  if (props.css)
+    console.warn(`Dialog.Content: Please use style instead of css
+    
+    This component is rendered in a portal and is outside
+    the scope of scope of design language.
+  `);
+
+  const {
+    dialogVisible,
+    setDialogVisibility,
+    triggerRect,
+    label,
+  } = React.useContext(context);
+
+  const [overlayVisible, setOverlayVisiblity] = React.useState(false);
+
+  React.useEffect(() => {
+    if (dialogVisible) setOverlayVisiblity(true);
```

Comment:
Put curly braces around body to make it less error prone

```suggestion
    if (dialogVisible) {
      setOverlayVisiblity(true);
    }
```